Docs: consistency for FQCN, spacing, etc

andrew-demb opened the pull request.

andrew-demb committed:
Docs: consistent spacing, consistent array-style, consistent FQCN, avoid double escaped slashes, avoid double quotes if not necessary

andrew-demb committed:
Docs: use special note block instead of markdown-based style

andrew-demb commented:
I found some more problems with a page about the second-level cache (https://www.doctrine-project.org/projects/doctrine-orm/en/2.10/reference/second-level-cache.html).
Probably not all of them I can fix because of a lack of knowledge about the RST format or how to find an alternative.

1) broken description for list elements (cache factories - https://www.doctrine-project.org/projects/doctrine-orm/en/2.10/reference/second-level-cache.html#cache-factory). Compiled docs contain a separated list with elements (interface) and 'imploded' description for them without any delimiters. 
2) all references to API docs target to 404 page (example - https://www.doctrine-project.org/api/orm/current/Doctrine/ORM/Cache/Region.html)
3) some places don't mention FQCN for referenced class or interface (within code snippets or inlined). Should we add FQCN for some of them? Example - https://www.doctrine-project.org/projects/doctrine-orm/en/2.10/reference/second-level-cache.html#cache-mode)
4) there is one link with an anchor to the same page, but in "compiled" docs it doesn't display proper label for it (example - https://www.doctrine-project.org/projects/doctrine-orm/en/2.10/reference/second-level-cache.html#entity-cache-definition)

greg0ire commented on docs/en/reference/second-level-cache.rst:
```diff
@@ -147,21 +147,21 @@ Built-in cached persisters
+    +-----------------------+--------------------------------------------------------------------------------------+
+    | Cache Usage           | Persister                                                                            |
+    +=======================+======================================================================================+
+    | READ_ONLY             | Doctrine\ORM\Cache\Persister\Entity\ReadOnlyCachedEntityPersister                    |
```
Github's RST rendering does not seem too happy with the removal of the backslashes… maybe we should use double backticks here?

andrew-demb replied:
Escaping leads to rendering double backslash on the website :(
 https://www.doctrine-project.org/projects/doctrine-orm/en/2.10/reference/second-level-cache.html#built-in-cached-persisters

andrew-demb replied:
Ok 👍. I will check it in few days

andrew-demb replied:
Seems like it works as expected

andrew-demb committed:
Docs: Quote FQCN in table with backticks to be compatible with all render engines

SenseException commented:
> 2. all references to API docs target to 404 page

API Docs were removed at some point. This link can be removed too.

> 3. Should we add FQCN for some of them?

If the code example is rather a running code than a descriptive code to display different approaches, the FQCN is welcome there.

andrew-demb committed:
Drop all mentions API doc - it is not available anymore

andrew-demb committed:
Add missed FQCN for code snippets

andrew-demb commented:
> all references to API docs target to 404 page
> API Docs were removed at some point. This link can be removed too.

OK. Removed.

> Should we add FQCN for some of them?
> If the code example is rather a running code than a descriptive code to display different approaches, the FQCN is welcome there.

Added to code examples.
I think examples will be more clear for now.

SenseException commented:
@andrew-demb Thank you for the improvements.
